src/cargo/util/toml/targets.rs

```diff
@@ -106,13 +105,10 @@ fn clean_lib(toml_lib: Option<&TomlLibTarget>,
             let legacy_path = package_root.join("src").join(format!("{}.rs", lib.name()));
             if legacy_path.exists() {
                 {
-                    let short_path = without_prefix(&legacy_path, package_root)
-                        .unwrap_or(&legacy_path);
-
                     warnings.push(format!(
                         "path `{}` was erroneously implicitly accepted for library `{}`,\n\
                          please rename the file to `src/lib.rs` or set lib.path in Cargo.toml",
-                        short_path.display(), lib.name()
+                        legacy_path.display(), lib.name()
                     ));
                 }
```

Comment:
I think we should be able to remove these `{ }`. They were needed before to constrain `short_path`'s scope (it borrows `legacy_path`, which is moved out of this block in the end), but looks like we don't need then anymore.